hubzero/hubzero-cms
# [PURR][#2585]Show hint to ORCID ID format and validate the ORCID input

PURR Ticket: https://purr.purdue.edu/support/ticket/2585

Validate the ORCID ID that publication submitter enters in ORCID text field

On edit author dialog or add new author dialog of publication submission process, the publication submitter can enter the ORCID ID for author. There is not any validation procedure to what the submitter enters, so it may cause the link to the author not properly set on publication page. Such as the submitter may enter "https://orcid.org/xxxx-xxxx-xxxx-xxxx", but we expect that the submitter enters only the 16-digit ORCID ID without the ORCID endpoint. 

The code changes include adding the ORCID ID structure hint in ORCID text field and validate whether what the submitter enters match the ORCID ID structure, which is a 16-digit string and the last position can be either a digit or 'X. 

Test:
1. Create a file publication, add two collaborators of the project to the publication.
2. Click the "edit" button to open the edit author dialog, where you are expected to see the "XXXX-XXXX-XXXX-XXXX" in ORCID text field and language "*Please enter the 16-digit ORCID ID" displays under the field.
3. If you enter invalid characters such as lowercase alphabet in the text box, it is expected to prompt alert message "*Invalid ORCID ID. Please enter the 16-digit ORCID ID".
4. When you enter a valid ORDCID ID, it is expected to prompt "*You have successfully entered a valid 16-digit ORCID ID".
5. On admin interface, it is expected to show  "XXXX-XXXX-XXXX-XXXX" in ORCID text field when you edit an author of a pending approval publication. And language "*ORCID ID is 16-digit" displays under the ORCID text field.

Test is done in local development environment and the functions work as expected.

Jerry
